drivers/sensor/bosch/bmp180/bmp180.c

```diff
@@ -0,0 +1,294 @@
+
+	if (i2c_burst_read_dt(&config->i2c, BMP180_REG_OUT_MSB, buf, sizeof(buf)) < 0) {
+		LOG_ERR(ERROR_MSG_FMT, "read", "temperature measurement");
+		return -EIO;
+	}
+
+	data->uncomp_temp = (int16_t)sys_get_be16(&buf[0]);
+
+	return 0;
+}
+
+static int bmp180_read_uncomp_press(const struct device *dev)
+{
+	const struct bmp180_config *config = dev->config;
+	struct bmp180_data *data = dev->data;
+	uint8_t buf[3];
+	uint8_t oversampling_setting = config->oversampling;
+	/* See Table 8 from the datasheet */
+	uint8_t ctrl_reg_val = BMP180_CMD_READ_PRESS + (oversampling_setting << 6);
+
+	if (i2c_reg_write_byte_dt(&config->i2c, BMP180_REG_CTRL_MEAS, ctrl_reg_val) < 0) {
+		LOG_ERR(ERROR_MSG_FMT, "start", "pressure measurement");
+		return -EIO;
+	}
+
+	/* Sleep for x ms based on the oversampling setting */
+	switch (oversampling_setting) {
+	case BMP180_MODE_ULTRA_LOW_POWER:
+		k_msleep(5);
+		break;
+	case BMP180_MODE_STANDARD:
+		k_msleep(8);
+		break;
+	case BMP180_MODE_HIGH_RESOLUTION:
+		k_msleep(14);
+		break;
+	case BMP180_MODE_ULTRA_HIGH_RESOLUTION:
+		k_msleep(26);
+		break;
+	default:
+		LOG_ERR("Invalid oversampling setting.");
+		return -EINVAL;
+	}
+
+	/* Read the MSB, LSB, and XLSB from the sensor */
+	if (i2c_burst_read_dt(&config->i2c, BMP180_REG_OUT_MSB, buf, sizeof(buf)) < 0) {
+		LOG_ERR(ERROR_MSG_FMT, "read", "pressure measurement");
+		return -EIO;
+	}
+
+	data->uncomp_press = (int32_t)sys_get_be24(buf);
+	data->uncomp_press >>= (8 - oversampling_setting);
+
+	return 0;
+}
+
+static int bmp180_read_temp(const struct device *dev)
+{
+	struct bmp180_data *data = dev->data;
+
+	if (bmp180_read_uncomp_temp(dev) < 0) {
+		return -EIO;
+	}
+
+	int32_t x1 = (data->uncomp_temp - (int32_t)data->ac6) * ((int32_t)data->ac5) >> 15;
+	int32_t x2 = ((int32_t)data->mc << 11) / (x1 + (int32_t)data->md);
+	int32_t b5 = x1 + x2;
+
+	data->temp = (b5 + 8) >> 4;
+
+	return 0;
+}
+
+static int bmp180_read_press(const struct device *dev)
+{
+	const struct bmp180_config *config = dev->config;
+	struct bmp180_data *data = dev->data;
+
+	if (bmp180_read_uncomp_temp(dev) < 0) {
+		return -EIO;
+	}
+
+	if (bmp180_read_uncomp_press(dev) < 0) {
+		return -EIO;
+	}
+
+	/* See datasheet, page 15 for more details */
+	int32_t x1 = (data->uncomp_temp - (int32_t)data->ac6) * ((int32_t)data->ac5) >> 15;
+	int32_t x2 = ((int32_t)data->mc << 11) / (x1 + (int32_t)data->md);
+	int32_t b5 = x1 + x2;
+
+	int32_t b6 = b5 - 4000;
+
+	x1 = (data->b2 * ((b6 * b6) >> 12)) >> 11;
+	x2 = (data->ac2 * b6) >> 11;
+	int32_t x3 = x1 + x2;
+	int32_t b3 = (((((int32_t)data->ac1) * 4 + x3) << config->oversampling) + 2) / 4;
+
+	x1 = (data->ac3 * b6) >> 13;
+	x2 = (data->b1 * ((b6 * b6) >> 12)) >> 16;
+	x3 = ((x1 + x2) + 2) >> 2;
+	uint32_t b4 = (data->ac4 * (uint32_t)(x3 + 32768)) >> 15;
+
+	uint32_t b7 = ((uint32_t)data->uncomp_press - b3) * (50000 >> config->oversampling);
+	int32_t p;
+
+	if (b7 < 0x80000000) {
+		p = (b7 * 2) / b4;
+	} else {
+		p = (b7 / b4) * 2;
+	}
+
```

Comment:
I wondered the if (b7< xx) statement here from the datasheet example
math wise it is exactly same  only a data overflow can happen if b7 is too big
so b7/b4 *2 would always be safe and no need for a if()